### PR TITLE
Fix notifications for flapping alerts

### PR DESCRIPTION
**Pull request by juliusv**
Fixes https://github.com/prometheus/alertmanager/issues/1063

### CODE REVIEW

**Comment by juliusv** [comment]
@brancz @josedonizetti

File: notify/notify.go [hunk 1]
```diff
@@ -561,6 +559,19 @@ func NewRetryStage(i Integration) *RetryStage {
 
 // Exec implements the Stage interface.
 func (r RetryStage) Exec(ctx context.Context, l log.Logger, alerts ...*types.Alert) (context.Context, []*types.Alert, error) {
+	// If we shouldn't send notifications for resolved alerts, but there are only
+	// resolved alerts, report them all as successfully notified (we still want the
+	// notification log to log them).
+	if !r.integration.conf.SendResolved() {
+		firing, ok := FiringAlerts(ctx)
+		if !ok {
+			return ctx, alerts, fmt.Errorf("firing alerts missing")
```
Comment (brancz) [inline notify/notify.go]: If we error out here the `MultiStage` will abort the further execution, therefore the resolved subset of this set of alerts will not be gossiped. My understanding is that if there are no firing alerts in this set of alerts, but there are resolved alerts, we still want to gossip that.

Comment (juliusv) [reply notify/notify.go]: Yep. OTOH isn't this the same thing we are doing here? https://github.com/prometheus/alertmanager/blob/master/notify/notify.go#L627

AFAICS if this errors it means there's a bug in the code that should be fixed, as the context key should always be set earlier in the pipeline?

Comment (brancz) [reply notify/notify.go]: You're absolutely right.

**Pull request merged**